module.go

```diff
@@ -371,6 +378,58 @@ func NewRPCMsgIn(r *http.Request, postbody []byte, code int, size int64, dur tim
 	}
 }
 
+// NewRPCMsgInWithModuleConfig creates a message from a ModuleConfig object
+// End-users of the golang module never need to use this
+// directly and it is only exposed for performance testing
+func NewRPCMsgInWithModuleConfig(mcfg *ModuleConfig, r *http.Request, postbody io.Reader) (*RPCMsgIn, error) {
+
+	now := time.Now()
+
+	// assemble a message to send to inspector
+	tlsProtocol := ""
+	tlsCipher := ""
+	scheme := "http"
+	if r.TLS != nil {
+		// convert golang/spec integers into something human readable
+		scheme = "https"
+		tlsProtocol = tlstext.Version(r.TLS.Version)
+		tlsCipher = tlstext.CipherSuite(r.TLS.CipherSuite)
+	}
+
+	// golang removes Host header from req.Header map and
+	// promotes it to r.Host field. Add it back as the first header.
+	hin := convertHeaders(r.Header)
+	if len(r.Host) > 0 {
+		hin = append([][2]string{{"Host", r.Host}}, hin...)
+	}
+
+	body, err := ioutil.ReadAll(postbody)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Looking at this further, it is probably best to just pass in the `[]byte` buffer directly to avoid the extra copy here. Unfortunately we make another copy with the `string(body)` below, too. Sorry about the back and forth on this. I was initially thinking we could eventually just pass through the `req.Body` reader, but we cannot due to having to copy the data and then put it back for the next handlers in the chain (https://github.com/signalsciences/sigsci-module-golang/pull/21/files#diff-e412f3417cba5c5d742615a6595e9a0dL214-L225)

Reply:
no problem about the back and forth, I'm just glad your taking the time the time to review it.